src/Buffer.h

```diff
@@ -473,6 +473,8 @@ class Buffer {
 
     /** Wrap a native handle, using the given device API. */
     int device_wrap_native(const DeviceAPI &d, uint64_t handle, const Target &t = get_jit_target_from_environment()) {
+        user_assert(d != DeviceAPI::Default_GPU) << "Cannot pass DeviceAPI::Default_GPU to device_wrap_native.\n";
+        user_assert(target_supports_device_api(t, d)) << "DeviceAPI " << d << " not supported by target " << t.to_string() << "\n";
```

Comment:
We should add this assert for all the calls to get_device_interface_for_device_api in Buffer.h. (It produces a much cleaner error than the nullptr assert down in HalideBuffer.h.)